crates/re_log_types/src/data_table_batcher.rs

```diff
@@ -0,0 +1,787 @@
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use crossbeam::channel::{Receiver, Sender};
+
+use crate::{DataRow, DataTable, SizeBytes, TableId};
+
+// ---
+
+/// Errors that can occur when creating/manipulating a [`DataTableBatcher`].
+#[derive(thiserror::Error, Debug)]
+pub enum DataTableBatcherError {
+    /// Error when parsing configuration from environment.
+    #[error("Failed to parse config: '{name}={value}': {err}")]
+    ParseConfig {
+        name: &'static str,
+        value: String,
+        err: Box<dyn std::error::Error + Send + Sync>,
+    },
+
+    /// Error spawning one of the background threads.
+    #[error("Failed to spawn background thread '{name}': {err}")]
+    SpawnThread {
+        name: &'static str,
+        err: Box<dyn std::error::Error + Send + Sync>,
+    },
+}
+
+pub type DataTableBatcherResult<T> = Result<T, DataTableBatcherError>;
+
+// ---
+
+/// Defines the the different thresholds of the associated [`DataTableBatcher`].
+///
+/// See [`Self::default`] and [`Self::from_env`].
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct DataTableBatcherConfig {
+    /// Duration of the periodic tick.
+    //
+    // NOTE: We use `std::time` directly because this library has to deal with `crossbeam` as well
+    // as std threads, which both expect standard types anyway.
+    //
+    // TODO(cmc): Add support for burst debouncing.
+    pub flush_tick: Duration,
+
+    /// Flush if the accumulated payload has a size in bytes equal or greater than this.
+    pub flush_num_bytes: u64,
```

Comment:
There's a subtle distinction between "batch size" and "flush threshold" -- I suspect they are related but it's not entirely explicit here. A bit more explanation could be helpful.

Reply:
I'm not sure what you're referring to? What's "batch size"? There's no config with that name :thinking:

Reply:
No, but this thing is a "Batcher" and so it produces batches and those batches have a size. In particular the thing I think I was curious about was whether the *batch* would have a size > flush_num_bytes (I believe the answer to this is yes), but I wanted to be explicit that when we flush, all of the outstanding bytes (including those above the flush threshold) will end up in the batch.

Reply:
Ah yes, that's what the `equal or greater than this` alludes to. I'll make it extra clear.